test/common/buffer/buffer_fuzz_test.cc

```diff
@@ -0,0 +1,378 @@
+#include <fcntl.h>
+#include <unistd.h>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+#include "common/common/stack_array.h"
+
+#include "test/common/buffer/buffer_fuzz.pb.h"
+#include "test/fuzz/fuzz_runner.h"
+
+#include "gtest/gtest.h"
+
+// Strong assertion that applies across all compilation modes and doesn't rely
+// on gtest, which only provides soft fails that don't trip oss-fuzz failures.
+#define FUZZ_ASSERT(x) RELEASE_ASSERT(x, "")
+
+namespace Envoy {
+
+namespace {
+
+// The number of buffers tracked. Each buffer fuzzer action references one or
+// more of these. We don't need a ton of buffers to capture the range of
+// possible behaviors, let's assume only 3 for now.
+constexpr uint32_t BufferCount = 3;
```

Comment:
I'm probably missing something, but why do we need this? Given enough runs, won't we do every combination even if each individual run uses a single buffer?

Reply:
I think we need at least two buffers, to test operations like `buffer1.move(buffer2)`.

Reply:
Ah OK, that makes sense. Maybe just add a quick comment to that effect?

Reply:
Yep, done.